main.go

```diff
@@ -2,110 +2,148 @@ package main
 
 import (
 	"context"
-	"net/http"
+	"flag"
+	"fmt"
 	"os"
 
+	producer "github.com/asecurityteam/component-producer"
 	"github.com/asecurityteam/ipam-facade/pkg/assetfetcher"
 	"github.com/asecurityteam/ipam-facade/pkg/assetstorer"
 	"github.com/asecurityteam/ipam-facade/pkg/domain"
 	v1 "github.com/asecurityteam/ipam-facade/pkg/handlers/v1"
 	"github.com/asecurityteam/ipam-facade/pkg/ipamfetcher"
+	"github.com/asecurityteam/ipam-facade/pkg/randomnumbergenerator"
 	"github.com/asecurityteam/ipam-facade/pkg/sqldb"
 	"github.com/asecurityteam/serverfull"
 	"github.com/asecurityteam/settings"
 )
 
-func main() {
-	ctx := context.Background()
-	source, err := settings.NewEnvSource(os.Environ())
+type producerConfig struct {
+	*producer.Config
+}
+
+func (*producerConfig) Name() string {
+	return "producer"
+}
+
+type config struct {
+	LambdaMode bool `description:"Use the Lambda SDK to start the system."`
+	Producer   *producerConfig
+	Postgres   *sqldb.PostgresConfig
+	Device42   *ipamfetcher.Device42ClientConfig
+}
+
+func (*config) Name() string {
+	return "ipamfacade"
+}
+
+type component struct {
+	Producer *producer.Component
+	Postgres *sqldb.PostgresComponent
+	Device42 *ipamfetcher.Device42ClientComponent
+}
+
+func (c *component) Settings() *config {
+	return &config{
+		LambdaMode: false,
+		Producer:   &producerConfig{c.Producer.Settings()},
+		Postgres:   c.Postgres.Settings(),
```

Comment:
suggestion: The postgresql component should probably be on selection of a storage backend for the data. This would, for example, make it a more seamless transition later if we decided to convert to DynamoDB global tables or the Aurora Serverless options later.

Reply:
Sure it should. I've been planning to break much of this code out into a `component-storage` lib for some time, but that is far outside the scope of this PR.